cmd/cmd.go

```diff
@@ -450,6 +450,28 @@ func (o *options) generateTemplate() (*template.Template, error) {
 			}
 			return obj, nil
 		},
+		"extractJSONParts": func(text string, part ...string) (interface{}, error) {
+			obj := make(map[string]interface{})
+			if err := json.Unmarshal([]byte(text), &obj); err != nil {
+				return "obj", err
+			}
+			parts := make([]string, 0)
+			for _, key := range part {
+				parts = append(parts, fmt.Sprintf("%v", obj[key]))
+			}
+			return strings.Join(parts, ", "), nil
+		},
+		"tryExtractJSONParts": func(text string, part ...string) interface{} {
+			obj := make(map[string]interface{})
+			if err := json.Unmarshal([]byte(text), &obj); err != nil {
+				return text
+			}
+			parts := make([]string, 0)
+			for _, key := range part {
+				parts = append(parts, fmt.Sprintf("%v", obj[key]))
+			}
+			return strings.Join(parts, ", ")
+		},
```

Comment:
How about introducing a helper function called `joinMap` that concatenates map values based on given keys?

```go
		"joinMap": func(m map[string]interface{}, sep string, keys ...string) string {
			var parts []string
			for _, key := range keys {
				parts = append(parts, fmt.Sprintf("%v", m[key]))
			}
			return strings.Join(parts, sep)
		},
```

Although the use of `joinMap` may be slightly longer than `tryExtractJSONParts`, it will offer greater reusability.

```
# joinMap
{{joinMap (.Message | tryParseJSON) ", " "message" "service.name" }}{{"\n"}}

# tryExtractJSONParts
{{tryExtractJSONParts .Message "message" "service.name"}}{{"\n"}}
```

Reply:
As you said, if it fails to parse as JSON, joinMap won't be possible to return the string as is. In this case, `tryExtractJSONParts` seems like a good option. 👍